app/src/main/java/org/oppia/app/utility/SplitScreenManager.kt

```diff
@@ -0,0 +1,71 @@
+package org.oppia.app.utility
+
+import android.content.Context
+import android.content.res.Resources
+import org.oppia.app.R
+import org.oppia.app.model.EphemeralQuestion
+import org.oppia.app.model.EphemeralState
+import org.oppia.app.player.state.StateFragment
+import org.oppia.app.topic.questionplayer.QuestionPlayerFragment
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+private const val MINIMUM_DIAGONAL_WIDTH = 7.0
+
+/**
+ * A helper class that is used to detect whether to split the screen in [QuestionPlayerFragment]
+ * and [StateFragment] or not based on multiple factors.
+ */
+@Singleton
+class SplitScreenManager @Inject constructor(private val context: Context) {
+  private val splitScreenInteractionIdsPool = listOf("DragAndDropSortInput", "ImageClickInput")
+
+  /**
+   * The actual function that decides whether to split or not.
+   *
+   * @param[interactionId] the id of the interaction of the state inside the [EphemeralState] or
+   * [EphemeralQuestion].
+   * @return `true` if the screen should be split, `false` otherwise.
+   */
+  fun isSplitPossible(interactionId: String): Boolean {
```

Comment:
Perhaps instead: ``shouldSplitScreen``? I'm assuming it's always 'possible', just not always desirable.

Reply:
You have different views then 😄 @rt4914 @BenHenning   
You can decide together what I should do.

Reply:
@MohamedMedhat1998 Use `shouldSplitScreen` as it makes sense after reading the ben's reason behind `possible` word.